### ISSUE TITLE
CI system sometimes pull older code than what's in the PR

### ISSUE THREAD

**Issue by sflxn** [vmware/vic#3897]
As a VIC engineer, I should expect the code pulled in a CI build matches the PR that kicked off the build.

Acceptance criteria:  CI no longer pulls in old code for PR builds.  Perhaps even display the commit sha in the build.

----

I have seen this multiple times.

Here is an example from the CI logs:

```

Full Name:	Test-Cases.Group1-Docker-Commands.1-29-Docker-Checkpoint.Docker checkpoint ls
Start / End / Elapsed:	20170209 18:36:17.888 / 20170209 18:36:17.912 / 00:00:00.024
Status:	FAIL (critical)
Message:	'docker: 'checkpoint' is not a docker command.
See 'docker --help'.' does not contain 'vSphere Integrated Containers does not yet implement checkpointing'
00:00:00.021KEYWORD ${rc}, ${output} = OperatingSystem . Run And Return Rc And Output docker %{VCH-PARAMS} checkpoint ls test-busybox
00:00:00.000KEYWORD BuiltIn . Should Be Equal As Integers ${rc}, 1
00:00:00.001KEYWORD BuiltIn . Should Contain ${output}, vSphere Integrated Containers does not yet implement checkpointing
Documentation:	
Fails if container does not contain item one or more times.
Start / End / Elapsed:	20170209 18:36:17.911 / 20170209 18:36:17.912 / 00:00:00.001
18:36:17.912	FAIL	'docker: 'checkpoint' is not a docker command.
See 'docker --help'.' does not contain 'vSphere Integrated Containers does not yet implement checkpointing'	
```

Notice, it attempts to call 'docker %{VCH-PARAMS} pull busybox'.  This was not the latest code in the PR.  This is the code in the PR:

```
Docker checkpoint ls
    ${rc}  ${output}=  Run And Return Rc And Output  docker1.13 %{VCH-PARAMS} checkpoint ls test-busybox
    Should Be Equal As Integers  ${rc}  1
    Should Contain  ${output}  vSphere Integrated Containers does not yet implement checkpointing
```

Notice the code in the PR calls 'docker1.13'.  Resubmitting the same build pulled in the latest code.  I have seen this issue before.  In the past, it may require more than one build resubmission.

**Comment by mhagen-vmware** [comment]
Why is this in To Do? It also needs to have a priority assigned.

**Comment by sflxn** [comment]
@mhagen-vmware I added acceptance criteria above.

**Comment by kreamyx** [comment]
In case this helps, when I faced the same issue, the only thing that made the most recent commit to get pulled was to rebase off of master (I guess merging would work too, not tested) but essentially it happens when there are conflicts

**Comment by zjs** [comment]
It's disappointing that this was marked `priority/low`. I wasted _hours_ investigating a test which was working locally, but consistently failing in drone. The root cause was that drone was not actually using the latest code.

> essentially it happens when there are conflicts

This may be a sufficient condition to cause this, but it is not necessary; I've experienced this issue on a branch which would merge cleanly.
